_overviews/toolkit/munit-assertions.md

```diff
@@ -1,18 +1,218 @@
-description: ???
+description: Comparing what you obtain with what you expect
 num: 4
 previous-page: munit-test-suite
 next-page: munit-exceptions
 ---
 
 {% include markdown.html path="_markdown/install-munit.md" %}
 
-<!--
-Tutorial about `assert`, `assertEquals`, `assertNotEquals` and `assertNoDiff`
+Assertions describe what to check in your tests. If any assertion in a unit test
+fails, then the test fails.
 
-https://scalameta.org/munit/docs/assertions.html
+Assuming you have defined a
+[test suite]({% link _overviews/toolkit/munit-test-suite.md %}) `MyTests`, this
+tutorial shows various ways to write assertions within test specifications.
 
-We explain clues in a later tutorial.
- -->
\ No newline at end of file
+{% tabs assertions-0 class=tabs-scala-version %}
+{% tab 'Scala 2' %}
+```scala
+class MyTests extends munit.FunSuite {
+  test("a unit test") {
+    // test specification goes here
+  }
+}
+```
+{% endtab %}
+{% tab 'Scala 3' %}
+```scala
+class MyTests extends munit.FunSuite:
+  test("a unit test") {
+    // test specification goes here
+  }
+```
+{% endtab %}
+{% endtabs %}
+
+## Comparing What You Get With What You Expect
+
+The most common form of assertions consists of comparing a result obtained from
+the tested program with an expected value that you know is correct. You achieve
+this with the method `assertEquals`, which you inherit from the test suite base
+class, `munit.FunSuite`.
+
+For instance, assuming your program provides a method `double`, which takes an
+integer value as parameter and returns that value multiplied by two, you can
+write the following assertion:
+
+{% tabs assertions-1 %}
+{% tab 'Scala 2 and 3' %}
+```scala
+assertEquals(double(21), 42)
+```
+{% endtab %}
+{% endtabs %}
+
+Note the order of the arguments: the first one is the result of the program, and
+the second one is the expected correct result. To understand why this is important,
+let us look at the test report you get in case the tested program is incorrect:
+
+```text
+# MyTests:
+# ==> X MyTests.some-unit-test  0.068s munit.ComparisonFailException: ./example/MyTests.test.scala:7
+# 6:  test("some-unit-test") {
+# 7:    assertEquals(double(21), 42)
+# 8:  }
+# values are not the same
+# => Obtained
+# 23
+# => Diff (- obtained, + expected)
+# -23
+# +42
+```
+
+The test report shows the difference between the obtained result (`23`) and the
+expected result (`42`).
+
+If, by mistake, you swap the order of the arguments of the call to `assertEquals`,
+you get a wrong test report:
+
+```text
+# MyTests:
+# ==> X MyTests.some-unit-test  0.068s munit.ComparisonFailException: ./example/MyTests.test.scala:7
+# 6:  test("some-unit-test") {
+# 7:    assertEquals(42, double(21)) // BAD EXAMPLE, don't copy this
+# 8:  }
+# values are not the same
+# => Obtained
+# 42
+# => Diff (- obtained, + expected)
+# -42
+# +23
+```
+
+As you can see, the test report is confusing because it says that the expected
+result is `23`, whereas it is actually `42`.
```

Comment:
While this is useful to note. I don't think we should expand on it with an example.

As pointed out by @szymon-rd in [this comment](https://github.com/scalacenter/docs.scala-lang/pull/7#issuecomment-1315283129) we should try to keep the tutorials as concise as possible. I agree with him, because verbosity gives the impression of complexity.

Reply:
Removed in 23521c6